### PR TITLE
git-delta: new, 0.18.2

### LINKED ISSUE #8464: pakreq: delta
Repo: https://github.com/dandavison/delta

Description: A syntax-highlighting pager for git, diff, and grep output

### PR DESCRIPTION
**Pull request by poscat0x04**
Topic Description
-----------------

Adds git-delta, closes #8464

Package(s) Affected
-------------------

- git-delta

Security Update?
----------------

No

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`

### CODE REVIEW

File: app-devel/git-delta/spec [hunk 1]
```diff
@@ -0,0 +1,4 @@
+VER=0.18.2
+SRCS="https://github.com/dandavison/delta/archive/refs/tags/$VER.tar.gz"
```
Comment (stdmnpkg) [inline app-devel/git-delta/spec]: use git instead of tarball

File: app-devel/git-delta/autobuild/beyond [hunk 2]
```diff
@@ -0,0 +1,14 @@
+install -vDm644 \
+  etc/completion/completion.bash \
```
Comment (stdmnpkg) [inline app-devel/git-delta/autobuild/beyond]: `"$SRCDIR"/etc/......`

File: app-devel/git-delta/autobuild/beyond [hunk 3]
```diff
@@ -0,0 +1,14 @@
+install -vDm644 \
```
Comment (stdmnpkg) [inline app-devel/git-delta/autobuild/beyond]: add some log, e.g. `abinfo "Installing shell completion ..."`

**Review by stdmnpkg — CHANGES_REQUESTED** [review]

**Comment by stdmnpkg** [comment]
is this package amd64 only?

**Comment by poscat0x04** [comment]
I've only tested it on x86, although I think it should work on every platform that its dependencies work.

**Comment by stdmnpkg** [comment]
https://buildit.aosc.io/pipelines/11803

failed on loongarch64, loongson3, ppc64el

looks like our mdbook is too old, updated in #8494

File: app-devel/git-delta/autobuild/beyond [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  "$SRCDIR"/etc/completion/completion.bash \
+  "$PKGDIR"/usr/share/bash-completion/completions/delta
+install -vDm644 \
+  "$SRCDIR"/etc/completion/completion.zsh \
+  "$PKGDIR"/usr/share/zsh/site-functions/delta
+install -vDm644 \
+  "$SRCDIR"/etc/completion/completion.fish \
+  "$PKGDIR"/usr/share/fish/vendor_completions.d/delta.fish
+
+cd manual
```
Comment (MingcongBai) [inline app-devel/git-delta/autobuild/beyond]: Use absolute path here.

File: app-devel/git-delta/autobuild/beyond [hunk 5]
```diff
@@ -0,0 +1,21 @@
+abinfo "Installing shell completion"
+
```
Comment (MingcongBai) [inline app-devel/git-delta/autobuild/beyond]: Drop this newline.

File: app-devel/git-delta/autobuild/beyond [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  "$PKGDIR"/usr/share/fish/vendor_completions.d/delta.fish
+
+cd manual
+
+abinfo "Building documentation"
+mdbook build --dest-dir html
+
+abinfo "Installing documentation"
+
+mkdir -p "$PKGDIR"/usr/share/doc/git-delta
```
Comment (MingcongBai) [inline app-devel/git-delta/autobuild/beyond]: Verbosity please.

File: app-devel/git-delta/autobuild/beyond [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+cp -r html "$PKGDIR"/usr/share/doc/git-delta
```
Comment (MingcongBai) [inline app-devel/git-delta/autobuild/beyond]: Verbosity and absolute path missing.

File: app-devel/git-delta/autobuild/beyond [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+  "$SRCDIR"/etc/completion/completion.fish \
+  "$PKGDIR"/usr/share/fish/vendor_completions.d/delta.fish
+
+cd manual
+
+abinfo "Building documentation"
+mdbook build --dest-dir html
+
+abinfo "Installing documentation"
+
```
Comment (MingcongBai) [inline app-devel/git-delta/autobuild/beyond]: Drop this newline.

File: app-devel/git-delta/autobuild/prepare [hunk 9]
```diff
@@ -0,0 +1 @@
+export RUSTONIG_SYSTEM_LIBONIG=1
```
Comment (MingcongBai) [inline app-devel/git-delta/autobuild/prepare]: Why was this needed? Please add a comment explaining why:

```
# Note: ...
```

File: app-devel/git-delta/spec [hunk 10]
```diff
@@ -0,0 +1,4 @@
+VER=0.18.2
+SRCS="git::commit=tags/$VER::https://github.com/dandavison/delta.git"
```
Comment (MingcongBai) [inline app-devel/git-delta/spec]: This package should be named `delta` with a `PKGPROV` to `git-delta`. We follow upstream project names by principle.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: app-devel/delta/autobuild/defines [hunk 11]
```diff
@@ -0,0 +1,12 @@
+PKGNAME=delta
+PKGPROV=git-delta
+PKGSEC=devel
+PKGDEP="glibc libgit2 zlib gcc-runtime onig"
+BUILDDEP="cargo pkg-config llvm mdbook"
+
+PKGDES="A syntax-highlighting pager for git, diff, grep, and blame output"
+
+# linker errors while building serde
```
Comment (stdmnpkg) [inline app-devel/delta/autobuild/defines]: annotate this with FIXME

```
# FIXME: linker errors while building serde
# (copy paste actual error message here so we can grep them when fix is available)
```

**Review by stdmnpkg — CHANGES_REQUESTED** [review]

**Review by stdmnpkg — APPROVED** [review]

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**